### PR TITLE
Tinkering

### PR DESCRIPTION
**Pull request by SignpostMarv**
rebased as requested by @jfschmakeit in #2

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+         verbose="true"
+         bootstrap="vendor/autoload.php">
+    <testsuites>
+        <testsuite name="php-photoslibrary Test Suite">
+            <directory>tests/Unit/V1/</directory>
+        </testsuite>
+    </testsuites>
+    <filter>
+        <whitelist>
+            <directory suffix=".php">src/Google/Photos/Library/V1/</directory>
```
Comment (jfschmakeit) [inline phpunit.xml.dist]: This also needs the `/Types` path I believe:
```
<directory suffix=".php">src/Google/Photos/Library/Types/</directory>
```

**Review by jfschmakeit — COMMENTED** [review]
Thanks for preparing this - looks good overall. Please add the additional directories and packages as well to the config.

I'm adding someone for a second set of eyes as well.

File: phpunit.xml.dist (hunk 1 above)
Comment (SignpostMarv) [reply phpunit.xml.dist]: can do- what about the GBPMetadata folder ?

File: composer.json [hunk 2]
```diff
@@ -8,14 +8,23 @@
   "minimum-stability": "stable",
   "autoload": {
     "psr-4": {
-      "": "src/",
+      "Google\\Photos\\Library\\V1\\": "src/Google/Photos/Library/V1/",
+      "GPBMetadata\\Google\\Photos\\Library\\V1\\": "src/GPBMetadata/Google/Photos/Library/V1/"
```
Comment (jfschmakeit) [inline composer.json]: Let's shorten these to `Google\\Photos\\` and `GPBMetadata\Google\Photos\` - this way the `Types` folders are included as well.

File: phpunit.xml.dist (hunk 1 above)
Comment (jfschmakeit) [reply phpunit.xml.dist]: We don't have any tests that touch the `GBPMetadata` folder, so I don't think we need to include it here.
However, it does sound like good practice to include to include both packages from the source directory. 
I think we can just shorten these to `.../Photos` - that way the `Types` and `Library.V1` are both included.

File: composer.json (hunk 2 above)
Comment (SignpostMarv) [reply composer.json]: took the slightly lazier option of shortening them to `"Google\\"` and `"GPBMetadata\\"` in b91bd1b

File: phpunit.xml.dist (hunk 1 above)
Comment (SignpostMarv) [reply phpunit.xml.dist]: as with composer.json, I have taken the slightly lazier option of shortening to `src/Google/` in 5bab386

**Comment by SignpostMarv** [comment]
popped in an editorconfig file as my editor's default settings conflicted with the style used in the repo.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,11 @@
+root = true
+
+[*]
```
Comment (jfschmakeit) [inline .editorconfig]: Please scope this to just PHP files: `[**.php]`

File: composer.json [hunk 4]
```diff
@@ -8,14 +8,23 @@
   "minimum-stability": "stable",
   "autoload": {
     "psr-4": {
-      "": "src/",
+      "Google\\": "src/Google/",
```
Comment (jfschmakeit) [inline composer.json]: Please change these to `src/Google/Photos` and `src/GPBMetadata/Photos` to be more tightly scoped. (Other client libraries are also using similar package names, so we should be more precise here.)

**Review by jfschmakeit — CHANGES_REQUESTED** [review]

File: composer.json (hunk 4 above)
Comment (SignpostMarv) [reply composer.json]: resolved in 2aa367c

File: composer.json [hunk 5]
```diff
@@ -8,14 +8,23 @@
   "minimum-stability": "stable",
   "autoload": {
     "psr-4": {
-      "": "src/",
+      "Google\\Photos\\": "src/Google/Photos/",
+      "GPBMetadata\\Photos\\": "src/GPBMetadata/Photos/"
```
Comment (jfschmakeit) [inline composer.json]: Ah, I had some trouble running this -  this needs to include `Google` in the name and path:
```
      "GPBMetadata\\Google\\Photos\\": "src/GPBMetadata/Google/Photos/"
```
(This should be last change, then we can merge it in. Appreciate you working through the comments!)

Comment (SignpostMarv) [reply composer.json]: whoops

will try to get that done tonight.

Comment (SignpostMarv) [reply composer.json]: resolved in 1ee0a3e

File: composer.json [hunk 6]
```diff
@@ -8,14 +8,23 @@
   "minimum-stability": "stable",
   "autoload": {
     "psr-4": {
-      "": "src/",
+      "Google\\Photos\\": "src/Google/Photos/",
+      "GPBMetadata\\Photos\\": "src/GPBMetadata/Google/Photos/"
```
Comment (jfschmakeit) [inline composer.json]: Not quite yet - this needs the full package name that's referenced in the path and the files within it, including `Google`.

So you'd want this:
```
      "GPBMetadata\\Google\\Photos\\": "src/GPBMetadata/Google/Photos/"
```

Comment (SignpostMarv) [reply composer.json]: would you object to me rebasing these commits so the merge isn't peppered with these corrections?

Comment (jfschmakeit) [reply composer.json]: Yes, feel free to squash these iterations together.  (I was going to squash them otherwise on final merge.)

Thanks again!

**Pull request merged**